### ISSUE TITLE
[bitnami/nginx] Adding extraServerBlockConfigmap via helm chart breaks setup of default server block.

### ISSUE THREAD

**Issue by fsniper** [bitnami/charts#13041]
### Name and Version

bitnami/nginx:1.22

### What steps will reproduce the bug?

- Deploy image to k8s via bitnami/nginx helm chart using `nginx.existingServerBlockConfigmap`


### What is the expected behavior?

Pod should be running healthy.
Deployment using 1.21 version of container image:
```
nginx 10:12:39.26                                                                                                                                                                                                                                          
nginx 10:12:39.26 Welcome to the Bitnami nginx container                                                                                                                                                                                                   
nginx 10:12:39.26 Subscribe to project updates by watching https://github.com/bitnami/containers                                                                                                                                                          
nginx 10:12:39.26 Submit issues and feature requests at https://github.com/bitnami/containers/issues                                                                                                                                                       
nginx 10:12:39.26                                                                                                                                                                                                                                          
nginx 10:12:39.26 INFO  ==> ** Starting NGINX setup **                                                                                                                                                                                                     
nginx 10:12:39.27 INFO  ==> Validating settings in NGINX_* env vars                                                                                                                                                                                        
nginx 10:12:39.28 WARN  ==> The NGINX configuration file '/opt/bitnami/nginx/conf/nginx.conf' is not writable by current user. Configurations based on environment variables will not be applied.                                                          │
nginx 10:12:39.29 INFO  ==> No custom scripts in /docker-entrypoint-initdb.d                                                                                                                                                                               
nginx 10:12:39.29 INFO  ==> Initializing NGINX                                                                                                                                                                                                             
realpath: /bitnami/nginx/conf/vhosts: No such file or directory                                                                                                                                                                                            
nginx 10:12:39.31 INFO  ==> ** NGINX setup finished! **                                                                                                                                                                                                                                                                                                                                                                                                                                                            
nginx 10:12:39.32 INFO  ==> ** Starting NGINX **                                                                                                                                                                                                           
```

### What do you see instead?

Deployment using 1.22 or 1.23:

```
nginx 10:57:56.01                                                                                                                                                                                                                                          
nginx 10:57:56.01 Welcome to the Bitnami nginx container                                                                                                                                                                                                   
nginx 10:57:56.01 Subscribe to project updates by watching https://github.com/bitnami/containers                                                                                                                                                           
nginx 10:57:56.01 Submit issues and feature requests at https://github.com/bitnami/containers/issues                                                                                                                                                       
nginx 10:57:56.01                                                                                                                                                                                                                                          
nginx 10:57:56.02 INFO  ==> ** Starting NGINX setup **                                                                                                                                                                                                     
nginx 10:57:56.03 INFO  ==> Validating settings in NGINX_* env vars                                                                                                                                                                                        
nginx 10:57:56.03 WARN  ==> The NGINX configuration file '/opt/bitnami/nginx/conf/nginx.conf' is not writable by current user. Configurations based on environment variables will not be applied.                                                          │
nginx 10:57:56.04 INFO  ==> No custom scripts in /docker-entrypoint-initdb.d                                                                                                                                                                               
cp: cannot create regular file '/opt/bitnami/nginx/conf/server_blocks/default-https-server-block.conf': Read-only file system
``

### Additional information

_No response_

**Comment by javsalgar** [comment]
Hi! 

Thank you so much for opening the ticket. Do you have an example configmap that triggers the issue?

**Comment by longkai** [comment]
@javsalgar I got the same issue, when you set both `containerPorts.https` and `serverBlock` you will got this error.

**Comment by fsniper** [comment]
@javsalgar @longkai I also have `containerPorts.https`

**Comment by fsniper** [comment]
This failure is on this block [1]
<img width="766" alt="image" src="https://user-images.githubusercontent.com/803734/197183012-c680775a-7f9b-4654-a6d5-1dd56b3fe688.png">
 and really depends on https port. If you have an extraServerBlockConfigmap this is mounted in as a readonly volume on 
https://github.com/bitnami/charts/blob/f48790b465021af15620f08d40fec2d065535e14/bitnami/nginx/templates/deployment.yaml#L213-L216 so this operation fails. 

Related code is here on the bitnami nginx container: [1] https://github.com/bitnami/containers/blob/cc2aa37d91335fe3238d9ef86efc0c6bf99e77ef/bitnami/nginx/1.22/debian-11/rootfs/opt/bitnami/scripts/nginx/setup.sh#L33

**Comment by rafariossaa** [comment]
Hi,
  Could you give it a try to the latest version ? I fixed the issue in container image: `1.23.2-debian-11-r2`

**Comment by fsniper** [comment]
@rafariossaa Thank you. I checked the image and it works for my use case. 

After checking the code here https://github.com/bitnami/containers/blob/main/bitnami/nginx/1.23/debian-11/rootfs/opt/bitnami/scripts/nginx/setup.sh#L34: 
<img width="1261" alt="image" src="https://user-images.githubusercontent.com/803734/197726838-59010d70-d1ff-4bfd-9f9a-dc203f13e479.png">

I realized this also disables the HTTPS default configuration in case of an extraServerBlock. Wouldn't this be breaking use cases where user wants a default HTTPs server but configures some extra server blocks?
